services/src/main/java/com/linkedin/kmf/services/TopicRebalancer.java

```diff
@@ -0,0 +1,427 @@
+   */
+  private RebalanceCondition monitoredTopicNeedsRebalance() {
+
+    ZkUtils zkUtils = createZooKeeperUtils();
+    List<PartitionInfo> partitionInfoList = partitionInfoForMonitoredTopic();
+    Collection<Broker> brokers = scala.collection.JavaConversions.asJavaCollection(zkUtils.getAllBrokersInCluster());
+    return monitoredTopicNeedsRebalance(partitionInfoList, brokers, _rebalanceThreshold);
+  }
+
+
+  private List<PartitionInfo> partitionInfoForMonitoredTopic() {
+    ZkUtils zkUtils = createZooKeeperUtils();
+    scala.collection.mutable.ArrayBuffer<String> topicList = new scala.collection.mutable.ArrayBuffer<>();
+    topicList.$plus$eq(_topic);
+    scala.collection.Map<Object, scala.collection.Seq<Object>> partitionAssignments =
+        zkUtils.getPartitionAssignmentForTopics(topicList).apply(_topic);
+    List<PartitionInfo> partitionInfoList = new ArrayList<>();
+    scala.collection.Iterator<scala.Tuple2<Object, scala.collection.Seq<Object>>> it = partitionAssignments.iterator();
+    while (it.hasNext()) {
+      scala.Tuple2<Object, scala.collection.Seq<Object>> scalaTuple = it.next();
+      Integer partition = (Integer) scalaTuple._1();
+      Node leader = new Node((Integer) zkUtils.getLeaderForPartition(_topic, partition).get(), "", -1);
+      Node[] replicas = new Node[scalaTuple._2().size()];
+      for (int i = 0; i < replicas.length; i++) {
+        Integer brokerId = (Integer) scalaTuple._2().apply(i);
+        replicas[i] = new Node(brokerId, "", -1);
+      }
+      partitionInfoList.add(new PartitionInfo(_topic, partition, leader, replicas, null));
+    }
+
+    return partitionInfoList;
+  }
+
+
+  /**
+   * This returns RebalanceCondition.OK unless one of the following conditions are met in this order:
+   * <ul>
+   *   <li> The minimum number of total monitored partitions falls below floor(brokers * partitionThreshold).
+   *        Returns PartitionsLow</li>
+   *   <li> One or more brokers does not have a monitored partition or falls below the minimum number of monitored
+   *        partitions. Returns BrokerUnderMonitored</li>
+   *   <li> One or more brokers is not assigned as a preferred leader for any partition.  BrokerUnderMonitored</li>
+   *   <li> One or more brokers is the currently elected leader of a monitored partition. BrokerNotLeader</li>
+   * </ul>
+   * @param partitionInfoList this should only contain partition info for the monitored topic
+   * @param brokers get this from ZkUtils, this should be all the active brokers in the cluster
+   * @param partitionThreshold the lower water mark for when we do not have enough monitored partitions
+   * @return see above
+   */
+  static RebalanceCondition monitoredTopicNeedsRebalance(List<PartitionInfo> partitionInfoList, Collection<Broker> brokers,
+      double partitionThreshold) {
+
+    int partitionCount = partitionInfoList.size();
+    int minPartitionCount = (int) (brokers.size() * partitionThreshold);
```

Comment:
Can you explain why we need to round the number of an integer here? I think user expects a rebalance if partition_num/broker_num < minimum_partition_to_broker_ratio, right?